fineract-investor/src/main/java/org/apache/fineract/investor/cob/loan/LoanAccountOwnerTransferBusinessStep.java

```diff
@@ -85,7 +87,13 @@ public Loan execute(Loan loan) {
             if (ExternalTransferStatus.PENDING.equals(transfer.getStatus())) {
                 handleSale(loan, settlementDate, transfer);
             } else if (ExternalTransferStatus.BUYBACK.equals(transfer.getStatus())) {
-                handleBuyback(loan, settlementDate, transfer);
+                try {
+                    handleBuyback(loan, settlementDate, transfer);
+                } catch (ExternalAssetOwnerTransferNotFoundException e) {
+                    log.error(e.getMessage());
```

Comment:
I would not use try catch for this. Please consider moving this logging and entry creation back to the `handleBuyback` method and instead of throwing an error, just execute these.

Also please dont forget you need send business events from the CANCELLED status change too!

Reply:
Done! code updated based on our conversation